cards/src/main/java/com/decathlon/vitamin/compose/cards/VitaminCards.kt

```diff
@@ -0,0 +1,258 @@
+import androidx.compose.ui.unit.dp
+import com.decathlon.vitamin.compose.buttons.VitaminButtons
+import com.decathlon.vitamin.compose.foundation.VitaminTheme
+import com.decathlon.vitamin.compose.foundation.vtmnLightColorPalette
+
+object VitaminCards {
+    /**
+     * A card with image at top
+     * @param painter the [Painter] resource for the image
+     * @param contentDescription to describe image for screen reader. Set to null only if image is decorative.
+     * @param content The content of the card. Can be a composable from [TopImageContent] or a custom composable.
+     * @param modifier The `Modifier` to be applied to the all card.
+     * @param shape to change the shape of the card.
+     * @param imageScale The scale [VitaminCardImageScale] of the image in the card. can be
+     * [VitaminCardImageScale.Fit] or
+     * [VitaminCardImageScale.Full]
+     * @param imageOverlayAlignment The overlay position if overlay is set. Default is [Alignment.TopStart]
+     * @param imageOverlay The overlay to set upon the image. No overlay by default.
+     */
+    @Composable
+    fun TopImage(
+        painter: Painter,
+        contentDescription: String?,
+        content: @Composable (TopImageContent.() -> Unit),
+        modifier: Modifier = Modifier,
+        shape: Shape = RoundedCornerShape(4.dp),
+        imageScale: VitaminCardImageScale = VitaminCardImageScale.Fit,
+        imageOverlayAlignment: Alignment = Alignment.TopStart,
+        imageOverlay: (@Composable () -> Unit)? = null,
+    ) = Card(modifier = modifier, shape = shape) {
+        Column {
+            CardImage(
+                painter = painter,
+                contentDescription = contentDescription,
+                modifier = Modifier.padding(if (imageScale == VitaminCardImageScale.Fit) 16.dp else 0.dp),
+                overlay = imageOverlay,
+                imageOverlayAlignment = imageOverlayAlignment
+            )
+            Box(
+                modifier = Modifier.padding(16.dp),
+            ) {
+                TopImageContent.content()
+            }
+        }
+    }
+
+    /**
+     * A card with full image in background
+     * @param painter the [Painter] resource for the image
+     * @param contentDescription to describe image for screen reader. Set to null only if image is decorative.
+     * @param content The content of the card. Can be a composable from [FullImageContent] or a custom composable.
+     * @param modifier The `Modifier` to be applied to the all card.
+     * @param shape to change the shape of the card.
+     * @param imageWithGradient A boolean value to add a gradient effect to the image. No gradient by default.
+     */
+    @Composable
+    fun FullImage(
+        painter: Painter,
+        contentDescription: String?,
+        content: @Composable (FullImageContent.() -> Unit),
+        modifier: Modifier = Modifier,
+        shape: Shape = RoundedCornerShape(4.dp),
+        imageWithGradient: Boolean = false,
```

Comment:
There is a variant with a gradient and another one with an black overlay. Maybe an enumeration is more appropriated here with: none, gradient and overlay?